lib/CMakeLists.txt

```diff
@@ -276,7 +276,9 @@ if(BUILD_SHARED_LIBS STREQUAL "ON")
     # Prefer shared libraries for sqlite3 and zlib
     add_library(sqlite3 SHARED IMPORTED GLOBAL)
     add_library(z SHARED IMPORTED GLOBAL)
-    target_link_libraries(mat PUBLIC sqlite3 PUBLIC z ${LIBS} "${CMAKE_THREAD_LIBS_INIT}" "${CMAKE_DL_LIBS}" "${CMAKE_REQUIRED_LIBRARIES}")
+    find_package(SQLite3 REQUIRED)
```

Comment:
> find_package(SQLite3 REQUIRED) [](http://example.com/codeflow?start=3&length=31)

Is there any scenario where the call to `find_package(SQLite3 REQUIRED)` is required? Assume the current hard-coded name still works, right?

Reply:
yes, the current hard-coded name still works. But I think the `find_package` is a preferred way by cmake, and it will report a clearer error if `sqlite3` is not installed